src/kaleidoscope/hardware/avr/AVRDevice.h

```diff
@@ -0,0 +1,50 @@
+/* -*- mode: c++ -*-
+ * hardware::avr::AVRDevice -- Generic AVR device base class
```

Comment:
This really should be an ATMega32U4Device, not an AVRDevice. Again, it's fine if it subclasses a generic AVRDevice.

Reply:
I'm ok with that, I suppose.